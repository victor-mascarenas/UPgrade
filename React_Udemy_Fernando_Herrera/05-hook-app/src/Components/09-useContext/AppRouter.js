import React from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import { AboutPage } from './AboutPage';
import { HomePage } from './HomePage';
import { LoginPage } from './LoginPage';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <BrowserRouter>
            <div>
                <NavBar/>
                <div className='container'>
                    <Routes>
                        <Route exact path="/" element={<HomePage/>}/>
                        <Route exact path="about" element={<AboutPage/>}/>
                        <Route exact path="login" element={<LoginPage/>}/>
                    </Routes>
                </div>
            </div>
        </BrowserRouter>
    );
};
