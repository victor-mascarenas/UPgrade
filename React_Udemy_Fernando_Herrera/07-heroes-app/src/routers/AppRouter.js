import React from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import { LoginPage } from '../components/login/LoginPage';

import { DashboardRouter } from './DashboardRouter';

export const AppRouter = () => {
    return (
        <BrowserRouter>
            <Routes>
                <Route path='/login' element={<LoginPage/>}/>
                <Route path='/*' element={<DashboardRouter/>}/>
            </Routes>
        </BrowserRouter>
    );
};
