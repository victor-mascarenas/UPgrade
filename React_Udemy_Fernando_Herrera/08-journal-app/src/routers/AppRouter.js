import React from 'react'
import { BrowserRouter, Route, Routes, Navigate } from 'react-router-dom'
import { JournalScreen } from '../components/journal/JournalScreen'
import { AuthRouter } from './AuthRouter'

export const AppRouter = () => {
    return (
        <BrowserRouter>
            <Routes>
                <Route path='/auth/*' element={<AuthRouter/>}/>
                <Route exact path='/' element={<JournalScreen/>}/>
                <Route path='*' element={<Navigate to='/auth/login'/>}/>
            </Routes>
        </BrowserRouter>
    );
}
