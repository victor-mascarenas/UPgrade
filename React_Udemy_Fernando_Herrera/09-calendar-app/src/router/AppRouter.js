import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux';
import { BrowserRouter, Navigate, Route, Routes } from 'react-router-dom';
import { startChecking } from '../actions/auth';
import { Login } from '../components/auth/Login';
import { CalendarScreen } from '../components/calendar/CalendarScreen';
import { PrivateRoutes } from './PrivateRoutes';
import { PublicRoutes } from './PublicRoutes';

export const AppRouter = () => {
    const dispatch = useDispatch();
    const {checking, uid} = useSelector(state => state.auth);
    useEffect(() => {
        dispatch(startChecking());
    }, [dispatch]);

    let returnValue;
    if (checking) {
        returnValue = <h5>Espere...</h5>;
    } else {
        returnValue = <BrowserRouter>
            <Routes>
                <Route exact path='/login' element={
                    <PublicRoutes isLoggedIn={!!uid}>
                        <Login/>
                    </PublicRoutes>
                }/>
                <Route exact path='/' element={
                    <PrivateRoutes isLoggedIn={!!uid}>
                        <CalendarScreen/>
                    </PrivateRoutes>
                }/>
                <Route path='/*' element={<Navigate to='/'/>}/>
            </Routes>
        </BrowserRouter>;
    }
    return returnValue;
};