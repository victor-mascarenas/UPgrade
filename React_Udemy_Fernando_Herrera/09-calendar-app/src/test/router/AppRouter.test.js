import React from 'react';
import {mount} from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { AppRouter } from '../../router/AppRouter';
// import { eventStartDelete } from '../../actions/events';

/* jest.mock('../../actions/events', () => ({
    eventStartDelete: jest.fn()
})); */

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
// store.dispatch = jest.fn();

describe('Pruebas en AppRouter', () => {
    test('Debe de mostrar el "espere"', () => {
        const initState = {
            auth: {
                checking: true
            }
        };
        const store = mockStore(initState);
        const wrapper = mount(
            <Provider store={store}>
                <AppRouter/>
            </Provider>
        );
        expect(wrapper).toMatchSnapshot();
    });
});