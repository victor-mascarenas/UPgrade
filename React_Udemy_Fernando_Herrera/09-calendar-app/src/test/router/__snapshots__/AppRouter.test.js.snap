// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AppRouter Debe de mostrar el "espere" 1`] = `
<h5>
  Espere...
</h5>
`;
